### PR TITLE
Add useTargetWidth as a prop to context menu component.

### PR DESCRIPTION
**Pull request by verona123**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Add support for passing the width to the context menu. 

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 1]
```diff
@@ -209,7 +210,7 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
           onDismiss={ this.props.onDismiss }
           bounds={ bounds }
           directionalHintFixed={ directionalHintFixed }>
-          <div ref={ (host: HTMLDivElement) => this._host = host } id={ id } className={ css('ms-ContextualMenu-container', className) }>
+          <div style={ { width: width ? width : 'auto'} } ref={ (host: HTMLDivElement) => this._host = host } id={ id } className={ css('ms-ContextualMenu-container', className) }>
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: only apply this style if user passes a width. we don't want to inline width:auto to every contextual menu out there. We might break someone's custom styles.

**Comment by verona123** [comment]
@MLoughry yes I am doing it as per the suggestion :)

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 2]
```diff
@@ -189,6 +190,11 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
     let hasCheckmarks = !!(items && items.some(item => !!item.canCheck));
     const submenuProps = this.state.expandedMenuItemKey ? this._getSubmenuProps() : null;
 
+    let contextMenuWidth;
+    if (useTargetWidth && this._target as HTMLElement) {
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: What if _target is not an `HTMLElement` but a `MouseEvent` can this happen or is the typing wrong?

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 3]
```diff
@@ -209,7 +215,7 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
           onDismiss={ this.props.onDismiss }
           bounds={ bounds }
           directionalHintFixed={ directionalHintFixed }>
-          <div ref={ (host: HTMLDivElement) => this._host = host } id={ id } className={ css('ms-ContextualMenu-container', className) }>
+          <div style={ { width: contextMenuWidth || 'auto'} } ref={ (host: HTMLDivElement) => this._host = host } id={ id } className={ css('ms-ContextualMenu-container', className) }>
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: nit, spacing after `'`

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: still don't want 'auto' inlined here if we don't need it. create a style object up above pass that in.

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: yup, probably safer to test the target to make sure you can grab a width from it

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx (hunk 2 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: yup, probably safer to test the target to make sure you can grab a width from it

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx (hunk 3 above)
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: I think if we just use `contetaulMenuWidth }` that should be fine.. an undefined value shouldn't be set

Comment (verona123) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: Doing this now

    let contextMenuStyle;
    if (useTargetWidth && this._target as HTMLElement) {

        let contextMenuWidth = (this._target as HTMLElement).offsetWidth;
        contextMenuStyle = {
            width: contextMenuWidth
        };
    }

Comment (verona123) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: style={contextMenuStyle}

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx (hunk 2 above)
Comment (verona123) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: How to i check that ?

File: common/changes/sayalis-contextMenuWidth_2017-05-23-22-52.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "added useTargetWidth prop to the context menu, when true it will use the target elemnt's width to render itself",
```
Comment (MLoughry) [inline common/changes/sayalis-contextMenuWidth_2017-05-23-22-52.json]: My fault for not be clearer about guidelines for this earlier; however, the general standard for these notes is `<affected component>: <short description from the point of view of the consumer of the package`. So, for example, 
```
ContextMenu: added `useTargetWidth` property to match the width of the anchor element when true
```

**Comment by MLoughry** [comment]
Also, the build is failing with the following TSLint failure: `Warning - tslint - src/components/ContextualMenu/ContextualMenu.tsx(202,1): error no-consecutive-blank-lines: Consecutive blank lines are forbidden`

File: common/changes/sayalis-contextMenuWidth_2017-05-23-22-52.json (hunk 4 above)
Comment (verona123) [inline common/changes/sayalis-contextMenuWidth_2017-05-23-22-52.json]: sure

**Comment by verona123** [comment]
ok, i will fix it

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 5]
```diff
@@ -190,11 +190,16 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
     let hasCheckmarks = !!(items && items.some(item => !!item.canCheck));
     const submenuProps = this.state.expandedMenuItemKey ? this._getSubmenuProps() : null;
 
-    let contextMenuWidth;
-    if (useTargetWidth && this._target as HTMLElement) {
-        contextMenuWidth = (this._target as HTMLElement).offsetWidth;
+    let contextMenuStyle;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: Please type this

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 6]
```diff
@@ -190,11 +190,16 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
...
+    let targetAsHtmlElement = this._target as HTMLElement;
+    if (useTargetWidth && targetAsHtmlElement && targetAsHtmlElement.offsetWidth) {
+        let contextMenuWidth = targetAsHtmlElement.offsetWidth;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: `const`

**Pull request merged**